src/wallet/coinselection.cpp

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <wallet/coinselection.h>
+#include <util.h>
+#include <utilmoneystr.h>
+
+// Descending order comparator
+struct {
+    bool operator()(const CInputCoin& a, const CInputCoin& b) const
+    {
+        return a.effective_value > b.effective_value;
+    }
+} descending;
+
+/*
+ * This is the Branch and Bound Coin Selection algorithm designed by Murch. It searches for an input set that can pay for the
```

Comment:
These long line doc strings don't fit on my screen, even when my browser is maximized :-(